Helper/Order.php

```diff
@@ -501,9 +495,20 @@ private function deleteRedundantQuotes($quote) {
      * @param Quote $quote
      */
     public function deactivateParentQuote($quote) {
-        $parentQuote = $this->cartHelper->getQuoteById($quote->getBoltParentQuoteId());
-        $parentQuote->setIsActive(false);
-        $this->cartHelper->saveQuote($parentQuote);
+        try {
+            $parentQuote = $this->cartHelper->getQuoteById($quote->getBoltParentQuoteId());
+            $parentQuote->setIsActive(false);
+            $this->cartHelper->saveQuote($parentQuote);
+        } catch (NoSuchEntityException $e) {
+            $this->bugsnag->registerCallback(function ($report) use ($quote) {
```

Comment:
here you're swallowing error. shouldn't we notifyExcepgtion here?

Reply:
@daisy1754 This method, deactivateParentQuote, as well as deleteRedundantQuotes, runs on initial order save (from frontend or hook) and every subsequent hook to ensure the status of quote/order in magento are as expected. Subsequent calls are there in case, from any unforeseen reason, the quotes are not cleared. For example, deleteRedundantQuotes, which is raw query, does not succeed due to some custom, store specific constraints. If error is thrown here in finding the quote means all is good and quote already deleted so we do not have / need anything to deactivate. After all, there is a bugsnag callback registered so we will have this info if exception is raised at a later point. I might add order increment id to it.